tasks/libs/junit_upload_core.py

```diff
@@ -42,7 +45,72 @@ def add_flavor_to_junitxml(xml_path: str, flavor: AgentFlavor):
     tree.write(xml_path)
 
 
-def split_junitxml(xml_path, codeowners, output_dir):
+def junit_upload_from_tgz(junit_tgz, codeowners_path=".github/CODEOWNERS"):
+    """
+    Upload all JUnit XML files contained in given tgz archive.
+    """
+    from codeowners import CodeOwners
+
+    with open(codeowners_path) as f:
+        codeowners = CodeOwners(f.read())
+
+    for test_file in TEST_OUTPUT_FILES:
+        if os.path.isfile(test_file):
+            with open(test_file) as f:
+                test_output = json.load(f)
+    else:
+        test_output = []
+    flaky_tests = [
+        "/".join([test["Package"], test["Test"]])
+        for test in test_output
+        if any(flake in test["Output"] for flake in FLAKE_MESSAGES)
```

Comment:
<!-- Nitpicks are small, trivial, but necessary changes. Distinguishing nitpick comments significantly helps direct the reader's attention to comments requiring more involvement. -->
**🥜 nitpick**
```suggestion
        if any(flakeMessage in test["Output"] for flakeMessage in FLAKE_MESSAGES)
```